shared/AppInsightsCore/src/JavaScriptSDK/EnvUtils.ts

```diff
@@ -354,3 +364,137 @@ export function sendCustomEvent(evtName: string, cfg?: any, customDetails?: any)
     }
     return false;
 }
+
+/**
+ * Redacts user information from a URL
+ * @param url - The URL string to redact
+ * @returns The URL with user information redacted
+ */
+function redactUserInfo(url: string): string {
+    const schemeEndIndex = url.indexOf(":");
+    if (schemeEndIndex === -1) {
+        // not a valid url
+        return url;
+    }
+    const len = url.length;
+    if (len <= schemeEndIndex + 2
+        || url.charAt(schemeEndIndex + 1) !== "/"
+        || url.charAt(schemeEndIndex + 2) !== "/") {
+        // has no authority component
+        return url;
+    }
+    // look for the end of the authority component:
+    //   '/', '?', '#' ==> start of path
+    let index: number;
+    let atIndex = -1;
+    for (index = schemeEndIndex + 3; index < len; index++) {
+        const c = url.charAt(index);
+
+        if (c === "@") {
+            atIndex = index;
+        }
+
+        if (c === "/" || c === "?" || c === "#") {
+            break;
+        }
+    }
+    if (atIndex === -1 || atIndex === len - 1) {
+        return url;
+    }
+    return url.substring(0, schemeEndIndex + 3) + "REDACTED:REDACTED" + url.substring(atIndex);
+}
+
+/**
+ * Redacts sensitive query parameters from a URL
+ * @param url - The URL string to redact
+ * @returns The URL with sensitive query parameters redacted
+ */
+function redactQueryParameters(url: string): string {
+    const questionMarkIndex = url.indexOf("?");
+    if (questionMarkIndex === -1 || !containsParamToRedact(url)) {
+        return url;
+    }
+    let urlAfterQuestionMark = "";
+    // To build a parameter name until we reach the '=' character
+    // If the parameter name is a one to redact, we will redact the value
+    let currentParamName = "";
+    for (let i = questionMarkIndex + 1; i < url.length; i++) {
```

Comment:
This can be slow, I thought we had some code (somewhere -- or it could just be that I've written code to do this a lot of time in the past and I'm mis-remembering) that handles the URL splitting... (this and command line parsing are very simular) -- but generally using indexOf "should" be faster as it's using native code for the iteration and comparison rather than a bunch of manipulation (but that also gets messy -- so I understand) -- let me search over the existing AI code so see if I can provide some examples.